### ISSUE TITLE
shareWithOptions broken?

### ISSUE THREAD

**Issue by arielweinberger** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#576]
Can anyone confirm that shareWithOptions is broken? Or it is not?

I ran the example script. on Android nothing happens.

Running iOS 9 and Android 5.1. Ionic Framework.

**Comment by arielweinberger** [comment]
So I managed to solve it.

If you install the plugin via `cordova plugin install cordova-plugin-x-socialsharing` you get version 5.0.X (where shareWithOptions is not available).

I removed the plugin and downloaded it using `vordova plugin install `https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin` and got version 5.0.1.

**Comment by EddyVerbruggen** [comment]
Yeah I just started working on `shareWithOptions` and it hasn't made it to a release yet. If you have any feedback on it already please feel free to share it.

**Comment by arielweinberger** [comment]
@EddyVerbruggen 
Basically it works like a charm. I think there should be an option to prioritize stuff on certain platforms.
So if for example the `files` array contains an image, I want to share it in a message or via Email (along with the message and the url, which is possible).

But WhatsApp or Facebook for example lets you share either a URL or an Image. So in this case the shareWithOptions was not useful. So it's like "alright, so I want to share an image, a URL and a message. On e-mails and messages that's fine. But on WhatsApp, if I do this, I am forced to only send the image or go back to using shareViaWhatsApp". I personally found the `shareWithOptions` method super useful and the whole plugin is amazing.

So bottom line, some way to prioritize img/url for certain apps (facebook, WhatsApp) would be awesome.
I hope I was clear.
